apollo-client/src/main/java/com/ctrip/framework/apollo/spring/auto/SpringValue.java

```diff
@@ -0,0 +1,105 @@
+package com.ctrip.framework.apollo.spring.auto;
+
+import com.ctrip.framework.apollo.util.function.Functions;
+import com.google.common.base.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Date;
+
+/**
+ * Spring @Value field or method info
+ * @author github.com/zhegexiaohuozi [wanghaomiao@huli.com]
+ * @since 2017/12/20.
+ */
+public class SpringValue {
+    private Object bean;
+    private Field field;
+    private Method method;
+    private boolean isField = false;
+    private String className;
+    private String fieldName;
+    private Function<String, ?> parser;
+    private Logger logger = LoggerFactory.getLogger(SpringValue.class);
+
+    private SpringValue(Object ins, Field field) {
+        this.bean = ins;
+        this.className = ins.getClass().getName();
+        this.fieldName = field.getName();
+        this.field = field;
+        this.isField = true;
+        this.parser = findParser(field.getType());
+    }
+
+    private SpringValue(Object ins, Method method) {
+        this.bean = ins;
+        this.method = method;
+        this.className = ins.getClass().getName();
+        this.fieldName = method.getName() + "(*)";
+        Class<?>[] paramTps = method.getParameterTypes();
+        if (paramTps.length != 1){
+            logger.error("invalid setter,can not update in {}.{}",className,fieldName);
+            return;
+        }
+        this.parser = findParser(paramTps[0]);
+    }
+
+    public static SpringValue create(Object ins, Field field){
+        return new SpringValue(ins,field);
+    }
+
+    public static SpringValue create(Object ins, Method method){
+        return new SpringValue(ins,method);
+    }
+
+    public void updateVal(String newVal){
+        try {
+            if (isField){
```

Comment:
if parser is null, I think we shouldn't do anything to the value?

Reply:
if parser is null,we can just consider it as string in apollo and other type must has it's parser as 
a convention.